controllers/awscluster_controller.go

```diff
@@ -380,6 +382,19 @@ func (r *AWSClusterReconciler) requeueAWSClusterForUnpausedCluster(log logr.Logg
 			return nil
 		}
 
+		awsCluster := &infrav1.AWSCluster{}
+		key := types.NamespacedName{Namespace: c.Spec.InfrastructureRef.Namespace, Name: c.Spec.InfrastructureRef.Name}
+
+		if err := r.Get(ctx, key, awsCluster); err != nil {
+			log.V(4).Error(err, "Failed to get AWS cluster")
+			panic(fmt.Sprintf("Failed to get AWS cluster %v", err))
```

Comment:
I think panicking here is probably not the way to go, as this could happen for example when the API is unavailable or if the Cluster resource has been created before the AWSCluster resource.

IMO, the solution here when there is an error is to log the error (as you've done) and fail closed (ie return nil). It means we miss the potential event from the Cluster object, but it's better than panicking IMO.

AFAIK the events from the cluster mapping are a bonus anyway, so should be safe to fail closed here